src/quilla/steps/validations/xpath.py

```diff
@@ -204,7 +210,50 @@ def _check_not_attribute_has_value(self) -> ValidationReport:
     def _check_visual_parity(self) -> ValidationReport:
         self._verify_parameters('baselineID')
 
-        return self._create_report(
-            False,
-            msg='VisualParity Validation State not yet implemented'
+        baseline_id = self.parameters['baselineID']
+
+        treatment_image_bytes = self.element.screenshot_as_png
+        treatment_image = Image.open(BytesIO(treatment_image_bytes))
+
+        plugin_result: List = self.ctx.pm.hook.quilla_get_visualparity_baseline(
+            ctx=self.ctx,
+            baseline_id=baseline_id
+        )
+
+        if len(plugin_result) == 0:
+            return VisualParityReport(
+                success=False,
+                target=self._target,
+                browser_name=self.driver.name,
+                baseline_id=baseline_id,
+                msg='No baseline storage mechanism configured, or no baseline found'
+            )
+
+        baseline_image_bytes, baseline_uri = plugin_result[0]
+        baseline_image = Image.open(BytesIO(baseline_image_bytes))
+
+        success = baseline_image == treatment_image  # Run the comparison with Pillow
+
+        if success:
+            return VisualParityReport(
+                success=success,
+                target=self._target,
+                browser_name=self.driver.name,
+                baseline_id=baseline_id,
+            )
+
+        treatment_uri = self.ctx.pm.hook.quilla_store_image(
+            ctx=self.ctx,
```

Comment:
Running ID needs to be passed in to help generate the correct treatment image path. If it's not available now, maybe add some comment.

Reply:
running ID will be attached to the Context when it is available